tools/codegen/codespec/merge_attributes.go

```diff
@@ -1,57 +1,103 @@
+	for _, attr := range mergedMap {
+		*existingAttrs = append(*existingAttrs, *attr)
+	}
+
+	sortAttributes(*existingAttrs)
+}
+
+func addOrUpdate(attr *Attribute, computability ComputedOptionalRequired, merged map[string]*Attribute, isFromResponse bool) {
+	if existingAttr, exists := merged[attr.Name.SnakeCase()]; exists {
+		if existingAttr.Description == nil || *existingAttr.Description == "" {
+			existingAttr.Description = attr.Description
+		}
+
+		if !isFromResponse && existingAttr.ComputedOptionalRequired != Required {
+			existingAttr.ComputedOptionalRequired = computability
+		}
+
+		if existingAttr.ListNested != nil && attr.ListNested != nil {
+			mergeNestedAttributes(&existingAttr.ListNested.NestedObject.Attributes, attr.ListNested.NestedObject.Attributes, computability, isFromResponse)
+		} else if attr.ListNested != nil {
+			existingAttr.ListNested = attr.ListNested
+		}
+
+		if existingAttr.SingleNested != nil && attr.SingleNested != nil {
+			mergeNestedAttributes(&existingAttr.SingleNested.NestedObject.Attributes, attr.SingleNested.NestedObject.Attributes, computability, isFromResponse)
+		} else if attr.SingleNested != nil {
+			existingAttr.SingleNested = attr.SingleNested
+		}
+
+		if existingAttr.SetNested != nil && attr.SetNested != nil {
+			mergeNestedAttributes(&existingAttr.SetNested.NestedObject.Attributes, attr.SetNested.NestedObject.Attributes, computability, isFromResponse)
+		} else if attr.SetNested != nil {
+			existingAttr.SetNested = attr.SetNested
+		}
+
+		if existingAttr.MapNested != nil && attr.MapNested != nil {
+			mergeNestedAttributes(&existingAttr.MapNested.NestedObject.Attributes, attr.MapNested.NestedObject.Attributes, computability, isFromResponse)
+		} else if attr.MapNested != nil {
+			existingAttr.MapNested = attr.MapNested
+		}
+	} else {
+		newAttr := *attr
+		newAttr.ComputedOptionalRequired = computability
+		merged[attr.Name.SnakeCase()] = &newAttr
+	}
+}
+
+func mergeAttributes(pathParams, createRequest, createResponse, readResponse Attributes) Attributes {
+	merged := make(map[string]*Attribute)
+
 	// Path parameters: all attributes will be "required"
 	for i := range pathParams {
-		addOrUpdate(&pathParams[i], Required)
+		addOrUpdate(&pathParams[i], Required, merged, false)
 	}
 
 	// POST request body: optional/required is as defined
 	for i := range createRequest {
-		addOrUpdate(&createRequest[i], createRequest[i].ComputedOptionalRequired)
+		addOrUpdate(&createRequest[i], createRequest[i].ComputedOptionalRequired, merged, false)
 	}
 
 	// POST/GET response body: properties not in the request body are "computed" or "computed_optional" (if a default is present)
 	for i := range createResponse {
-		if _, exists := merged[createResponse[i].Name.SnakeCase()]; !exists {
-			if isOptional(&createResponse[i]) {
-				addOrUpdate(&createResponse[i], ComputedOptional)
-			} else {
-				addOrUpdate(&createResponse[i], Computed)
-			}
+		if isOptional(&createResponse[i]) {
```

Comment:
```suggestion
		if hasDefault(&createResponse[i]) {
```
would this be more accurate?

Reply:
yes, updated